install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -20,9 +20,10 @@ spec:
       rules:
         - alert: MCDDrainError
           expr: |
-            mcd_drain > 0
+            mcd_drain_err > 0
```

Comment:
How does this react to labels changing?  Does that reset the clock?  Do we alert if at least one node has been reporting an error for every time in the past 15m, or does it need a single node to report an error for the full 15m?  I've reread [the alert docs][1], but am still not clear on this :/

[1]: https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/

Reply:
Nothing here has changed how our alerts fundamentally work. 

These are alerts from mcd which are per node. So any mcd alert is based on the status of the underlying node. 

In this case if the drain error for the specific node is still present after the 15 minutes (which means failing after many, many retries) it will fire instead of immediately firing after very the first failure. This means that we have a better signal as to whether or not the error is transient/self-recovering before we send anything.

If another node somehow also started failing to drain, there would be a separate alert associated with that node, etc...

We plan on adding more in the future this is intended to fix current behavior.